pkg/config/expand.go

```diff
@@ -143,9 +144,24 @@ func (bp Blueprint) expandGroup(gp groupPath, g *Group) error {
 func (bp Blueprint) expandModule(mp ModulePath, m *Module) error {
 	bp.applyUseModules(m)
 	bp.applyGlobalVarsInModule(m)
+	// Versioned Module Logic for Embedded Modules
+	if bp.ToolkitModulesURL != "" && bp.ToolkitModulesVersion != "" {
+		if strings.HasPrefix(m.Source, "modules/") || strings.HasPrefix(m.Source, "community/") {
+			newSource, err := constructVersionedModuleSource(bp.ToolkitModulesURL, m.Source, bp.ToolkitModulesVersion)
+			if err != nil {
+				return fmt.Errorf("error constructing versioned module source: %w", err)
+			}
+			m.Source = newSource
+		}
+	}
 	return validateModuleInputs(mp, *m, bp)
 }
 
+// TODO: Add validation and error checks for baseURL and version
```

Comment:
nit. This looks like a bad place to do validations. `ToolkitModulesURL` and `ToolkitModulesVersion` should be validated in advance, not during module processing.

You can save 5 lines by removing never-error handling